open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataEventConsumer.java

```diff
@@ -514,15 +509,52 @@ private void stopRunning()
     }
 
 
-    private class HandleRebalance implements ConsumerRebalanceListener
-    {
+    private class HandleRebalance implements ConsumerRebalanceListener {
         AuditLog auditLog = null;
+
         public HandleRebalance(AuditLog auditLog) {
             this.auditLog = auditLog;
         }
 
-        public void onPartitionsAssigned(Collection<TopicPartition> partitions)
-        {
+        public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
+
+            // Check if we need to rewind to handle initial startup case -- but only on first assignment
+            try {
+                if (initialPartitionAssignment) {
+                    log.info("Received initial rebalance event");
+
+                    long partitionCount = partitions.size();
+
+                    if (partitionCount != 1) {
+                        log.info("Received rebalance event with " + partitionCount + " partitions.  This is not supported.");
```

Comment:
I suggest warning - as this means that Kafka is not usable until we get a partitionCount of 1. 
Could the partitionCount be 0?
If there is more than one partition - could we use the first one ? Or would that not work for some reason?

Reply:
With our current topic connector I cannot see how this would be anything other than 1.
It is possible to subscribe to multipe topics, partitions through a single poll, but we don't do this.
Just a bit of defensive coding given the complexity of kafka - and possibility of changes down the line
This event is really just of interest to us, not critical -- so a poll() will require a partition before it can work, and if it fails we'll see errors elsewhere
Hence I'm inclined to leave as is